CMakeLists.txt

```diff
@@ -8,6 +8,12 @@ if(NOT CMAKE_CONFIGURATION_TYPES AND NOT CMAKE_BUILD_TYPE)
   set(CMAKE_BUILD_TYPE RelWithDebInfo)
 endif()
 
+option(WITH_ASAN "Compile with address sanitizer support" OFF)
+if(WITH_ASAN)
+  add_compile_options(-fsanitize=address)
+  add_link_options(-fsanitize=address)
+endif()
```

Comment:
I am not a big fan of setting these flags globally, especially for libraries. I think
```
target_compile_options(urcl PRIVATE -fsanitize=address)
target_link_options(urcl PRIVATE -fsanitize=address)
```
would be the better choice.

Reply:
In principle I would agree. However, using this inside the main CMakeLists.txt
   ```cmake
   ...
   add_library(ur_client_library::urcl ALIAS urcl)
   if(WITH_ASAN)
     target_compile_options(urcl PRIVATE -fsanitize=address)
     target_link_options(urcl PRIVATE -fsanitize=address)
   endif()
   target_compile_options(urcl PRIVATE -Wall -Wextra -Wno-unused-parameter)
   target_compile_options(urcl PUBLIC ${CXX17_FLAG})
   ...
   ```
leads to this (without further modifications)
   ```
   build/examples/dashboard_example 192.168.56.101
   ==416292==ASan runtime does not come first in initial library list; you should either link runtime to your application or manually preload it with LD_PRELOAD.
   ```
So, one has to add the targeted sanitizer to **each** target, as the targets depending on urcl won't work (won't run) otherwise. Since from a maintenance perspective it seems not right to copy those four lines to each of our many targets I would suggest sticking with the global flag. It should only be activated manually for testing purposes, so I guess, this would be valid price to pay.

Reply:
Can you try using
```
  target_compile_options(urcl PUBLIC -fsanitize=address)
  target_link_options(urcl PUBLIC -fsanitize=address)
```
instead? From my understanding that should serve exactly this role, e.g. propagating to all targets linking to `urcl`.

Reply:
Sounds like a sane idea, will try that.